protocol_test.go

```diff
@@ -1,50 +1,53 @@
-package main
+package rpc
 
 import (
 	"fmt"
+	"io"
 	"net"
-	"os"
 	"testing"
-	"time"
 
-	rpc "github.com/keybase/go-framed-msgpack-rpc"
 	"github.com/stretchr/testify/assert"
 	"golang.org/x/net/context"
 )
 
 var testPort int = 8089
 
-func TestMain(m *testing.M) {
-	err := prepServer()
-	if err != nil {
-		fmt.Println("A server error occurred")
-		os.Exit(-1)
-	}
-	os.Exit(m.Run())
-}
-
-func prepServer() error {
-	server := &Server{port: testPort}
+func prepServer(listener chan error) error {
+	server := &server{port: testPort}
 
 	serverReady := make(chan struct{})
 	var err error
 	go func() {
-		err = server.Run(serverReady)
+		err = server.Run(serverReady, listener)
```

Comment:
There's a race here -- server.Run closes serverReady before it returns, so err is in an indeterminate state when it's returned